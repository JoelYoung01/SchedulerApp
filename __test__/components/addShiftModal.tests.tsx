import React, { useEffect } from "react";
import ReactDOM from "react-dom";
import { render, fireEvent } from "@testing-library/react";
