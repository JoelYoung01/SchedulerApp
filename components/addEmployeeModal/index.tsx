export * from "./AddEmployeeModal";
export * from "./AvailiabilityTabs";
export * from "./AvailabilityEditor";
export * from "./EditEmployeeInfo";
export * from "./TabPanel";
