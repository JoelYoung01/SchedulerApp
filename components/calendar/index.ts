export * from "./Calendar";
export * from "./DailyShifts";
export * from "./ShiftCard";
export * from "./ShiftInfo";
export * from "./WeeklyDate";
