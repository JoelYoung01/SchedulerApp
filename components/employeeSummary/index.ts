export * from "./EmployeeSummary";
export * from "./EmployeeCard";
