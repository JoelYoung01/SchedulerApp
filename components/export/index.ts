export * from "./ExportModal";
export * from "./FormatSelect";
