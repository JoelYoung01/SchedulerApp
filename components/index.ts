export * from "./calendar";
export * from "./export";
export * from "./layout";
