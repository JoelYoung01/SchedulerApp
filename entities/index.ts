export * from "./shift";
export * from "./employee";
export * from "./time";
export * from "./types";
export * from "./schedule";
export * from "./color";
