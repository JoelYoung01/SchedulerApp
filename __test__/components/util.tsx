import { expect } from "@jest/globals";
import type { MatcherFunction } from "expect";

import "@testing-library/jest-dom";
import { render, screen, fireEvent, Matcher } from "@testing-library/react";

import React from "react";
import { ExportModal, ExportType } from "../../components";

const toBeHidden: MatcherFunction<[]> = function toBeHidden(actual) {
  if (!(actual instanceof HTMLElement)) {
    throw new Error("actual must be an HTMLElement");
  }
  let current: HTMLElement | null = actual;
  while (current !== null) {
    if (current.style.display === "none") {
      return {
        message: () =>
          "Expected element or parent not to be hidden by `style.display = 'none'`",
        pass: true
      };
    }
    current = current.parentElement;
  }
  return {
    message: () =>
      "Expected element or parent to be hidden by `style.display = 'none'`",
    pass: false
  };
};
expect.extend({ toBeHidden });

declare module "expect" {
  interface AsymmetricMatchers {
    toBeHidden: () => void;
  }
  interface Matchers<R> {
    toBeHidden: () => R;
  }
}

export function checkExportHide(
  name: string,
  element: JSX.Element,
  selector: Matcher
): void {
  test(name, async() => {
    const exportRef: React.RefObject<React.ReactInstance> = React.createRef();
    // Need to add export modal to add printing events
    render(
      <>
        {element}
        <ExportModal
          componentToExport={exportRef}
          defaultValue={ExportType.pdf}
        />
        <div className="printed" data-testid="shown" />
      </>
    );
    expect(screen.getByText(selector)).not.toBeHidden();
    fireEvent(window, new Event("beforeprint"));
    expect(screen.getByText(selector)).toBeHidden();
    expect(screen.getByTestId("shown")).not.toBeHidden();
    fireEvent(window, new Event("afterprint"));
    expect(screen.getByText(selector)).not.toBeHidden();
  });
}
