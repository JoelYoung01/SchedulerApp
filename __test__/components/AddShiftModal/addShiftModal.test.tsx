import "@testing-library/jest-dom";
import React from "react";
import { render, waitFor } from "@testing-library/react";
import { AddShiftModal } from "../../../components";
import userEvent from "@testing-library/user-event";
import { DayOftheWeek, Shift, Time } from "../../../entities";

test("Add Shift Renders", () => {
  const addShiftModal = render(
    <AddShiftModal existingShifts={[]} dispatch={jest.fn()} />
  );
  expect(addShiftModal).toMatchSnapshot();
});

test("Modal opens and closes", async () => {
  const user = userEvent.setup();
  const dispatch = jest.fn();
  dispatch.mockResolvedValue(undefined);
  const modal = render(
    <AddShiftModal existingShifts={[]} dispatch={dispatch} />
  );

  // Click the modal button
  const openButton = modal.getByText(/ADD SHIFT/i);
  await user.click(openButton);

  // Make sure the modal exists
  {
    const header = modal.queryByText(/ADD SHIFT/i);
    expect(header).not.toBe(null);

    expect(modal).toMatchSnapshot();
  }

  // Click the submit button
  await user.click(modal.getByText(/Close/i));
});

test("Adding Shifts", async () => {
  // Setup modal, open it
  const user = userEvent.setup();
  const dispatch = jest.fn();
  dispatch.mockResolvedValue(undefined);
  const modal = render(
    <AddShiftModal existingShifts={[]} dispatch={dispatch} />
  );
  await user.click(modal.getByText(/Add Shift/i));

  expect(modal.getByText(/Submit/i)).toBeDisabled();

  // Enter shift information
  await user.click(modal.getByLabelText(/Job Title/i));
  await user.click(modal.getByLabelText(/Select Start Day/i));
  await user.click(modal.getByText(/Sunday/i));
  await user.type(modal.getByLabelText(/Select Start Time/i), "10:10 PM");
  await user.type(modal.getByLabelText(/Select End Time/i), "02:10 AM");

  // Clicking Submit button
  await waitFor(() => expect(modal.getByText(/Submit/i)).not.toBeDisabled());
  await user.click(modal.getByText(/Submit/i));

  // Assert
  const newShift = new Shift(
    "",
    Time.fromString("22:10", DayOftheWeek.Sunday),
    Time.fromString("2:10", DayOftheWeek.Monday)
  );
  expect(dispatch.mock.calls).toEqual([[{ add: newShift }]]);
});
