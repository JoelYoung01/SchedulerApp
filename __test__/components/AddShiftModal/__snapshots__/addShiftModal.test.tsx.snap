// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Shift Renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Add Shift
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Add Shift
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Modal opens and closes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Add Shift
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        >
          <span
            class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
            style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
          >
            <span
              class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
            />
          </span>
        </span>
      </button>
    </div>
    <div
      aria-describedby="addShiftDescription"
      aria-labelledby="addShiftTitle"
      class="MuiModal-root css-79ws1d-MuiModal-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-t1klkj-MuiPaper-root-MuiCard-root"
        tabindex="-1"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            id="shift-modal-title"
          >
            Add a Shift
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ix1f03-MuiTypography-root"
            id="modal-StartTime"
          >
             
          </p>
          <div
            class="MuiBox-root css-ece9u5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r0:"
                  id=":r0:-label"
                >
                  Shift Name
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r0:"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Shift Name
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ix1f03-MuiTypography-root"
            id="modal-StartTime"
          >
             
          </p>
          <div
            class="MuiBox-root css-ece9u5"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                id="day-label"
              >
                Select Start Day
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="day-label"
                  class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    â€‹
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Select Start Day
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ix1f03-MuiTypography-root"
            id="modal-StartTime"
          >
             
          </p>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r1:"
              id=":r1:-label"
            >
              Select Start Time
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                placeholder="hh:mm (a|p)m"
                type="tel"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="Choose time"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ClockIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                    <path
                      d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Select Start Time
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ix1f03-MuiTypography-root"
            id="modal-EndTime"
          />
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for=":r2:"
              id=":r2:-label"
            >
              Select End Time
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r2:"
                placeholder="hh:mm (a|p)m"
                type="tel"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="Choose time"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ClockIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                    <path
                      d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Select End Time
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ix1f03-MuiTypography-root"
            id="modal-submit"
          />
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium css-rpzv7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Close
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <span
            aria-label="Start Day not set, Start Time not set, End Time not set"
            class=""
            data-mui-internal-clone-element="true"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              Submit
            </button>
          </span>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Add Shift
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      >
        <span
          class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
